tests/utils.py

```diff
@@ -205,7 +206,17 @@ def assert_ldd_version_container(container: Container, version: str) -> None:
 
 
 def snapshot_pid_profile(profiler: ProfilerInterface, pid: int) -> ProfileData:
-    return profiler.snapshot()[pid]
+    last_snapshot = profiler.snapshot()
+
+    def has_profile() -> bool:
+        nonlocal last_snapshot
+        if pid in last_snapshot:
+            return True
+        last_snapshot = profiler.snapshot()
+        return pid in last_snapshot
+
+    wait_event(timeout=5, stop_event=Event(), condition=has_profile, interval=0.1)
```

Comment:
So this will try to `snapshot()` every 100ms? I don't think we can do snapshots that fast, it is a very heavy operation. If the issue is that pids do not appear fast enough in snapshots (what is the underlying reason 🤔 ? I see it's independent of runtime) then I'd be more comfortable sticking a `sleep(2)` before `profiler.snapshot()` instead of calling `snapshot` multiple times in a row.